Autopilot/Test/Src/DataStructures/Test_Vector_Class.cpp

```diff
@@ -0,0 +1,109 @@
+#include <gtest/gtest.h>
+#include "math.h"
+#include "vectorClass.hpp"
+
+
+using namespace std; 
+using ::testing::Test;
+
+TEST(VectorClassTest, DefaultConstructor){
+    /***SETUP***/
+    Vector3D vector;
+    /***ASSERTS***/
+    EXPECT_EQ(vector.getX(), 0);
+    EXPECT_EQ(vector.getY(), 0);
+    EXPECT_EQ(vector.getZ(), 0);
+}
+
+TEST(VectorClassTest, DefinedConstructor){
+    /***SETUP***/
+    Vector3D vector(1,1,1);
+    /***ASSERTS***/
+    EXPECT_EQ(vector.getX(), 1);
+    EXPECT_EQ(vector.getY(), 1);
+    EXPECT_EQ(vector.getZ(), 1);
+}
+
+TEST(VectorClassTest, SettersAndGetters){
+    /***SETUP***/
+    Vector3D vector;
+    vector.setX(5);
+    vector.setY(5);
+    vector.setZ(5);
```

Comment:
ngl, for tests like this it's better to set each one of them to different values (ex. 1, 2, 3) so you are 100% sure the y value was not set to the x value and stuff. Just makes tests more secure

Reply:
Seems stupid I know, but it's good practice and may as well apply it here since it's an easy fix

Reply:
for sure, i'll change that